spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -500,20 +553,68 @@ function chanCloseConfirm(
     connection = provableStore.get(connectionPath(channel.connectionHops[0]))
     abortTransactionUnless(connection !== null)
     abortTransactionUnless(connection.state === OPEN)
+
+    let counterpartyHops: []string = []
+    if (connectionHops.length > 1) {
+      counterpartyHops = getCounterPartyHops(proofInit)
+    } else {
+      counterpartyHops = [connection.counterpartyConnectionIdentifier]
+    }
+
     expected = ChannelEnd{CLOSED, channel.order, portIdentifier,
-                          channelIdentifier, [connection.counterpartyConnectionIdentifier], channel.version}
-    abortTransactionUnless(connection.verifyChannelState(
-      proofHeight,
-      proofInit,
-      channel.counterpartyPortIdentifier,
-      channel.counterpartyChannelIdentifier,
-      expected
-    ))
+                          channelIdentifier, counterpartyHops, channel.version}
+
+    if (connectionHops.length > 1) {
+      consensusState = provableStore.get(consensusStatePath(connection.ClientId, proofHeight))
+      key = host.ChannelPath(counterparty.PortId, counterparty.ChannelId)
+      counterpartyConnectionEnd = abortTransactionUnless(getMultihopConnectionEnd(proofInit))
+      prefix = counterpartyConnectionEnd.GetCounterparty().GetPrefix()
+
+      abortTransactionUnless(VerifyMultihopProof(
```

Comment:
nit: Seems like the logic is the same for each method, maybe we should refactor the spec into a separate fn? Optional though, I don't feel super strongly about it either way.

Reply:
Yeah, agree, I'll make a note of it for the next round of updates.